addon/m3-tracked-array.js

```diff
@@ -25,20 +25,35 @@ export default class M3TrackedArray extends ArrayProxy {
   }
 
   replaceContent(idx, removeAmt, newItems) {
-    newItems = newItems.map((item, idx) => {
+    // Update childModelDatas array
+    // mapping to array of nested models
+    const childModelDatas = this._model._internalModel._modelData.__childModelDatas;
+    const key = this._key;
+    if (childModelDatas && childModelDatas[key] && Array.isArray(childModelDatas[key])) {
+      const newItemsInChildModelData = new Array(newItems.length);
+      Array.prototype.splice.apply(
+        childModelDatas[key],
+        [idx, removeAmt].concat(newItemsInChildModelData)
+      );
+    }
```

Comment:
I think it's confusing to have tracked array know so much about the internals of modeldata.

I suspect there's something better we can do, but for now let's do something like add `modelData._resizeChildModelData(idx,removeLength,addLength)`  so model data can own the logic here rather than tracking array know about its internals

Reply:
Moved the implementation to `modelData._resizeChildModelData`